hrjobcontract/CRM/Hrjobcontract/Export/Export.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class CRM_Hrjobcontract_Export_Export {
+
+  /**
+   * This method is basing on CRM_Export_BAO_Export::writeCSVFromTable()
+   * but adds check for hrjobcontract entities and runs values converting
+   * on them so we get proper Job Contract entity values exported.
+   *
+   * @param $exportTempTable
+   * @param $headerRows
+   * @param $sqlColumns
+   * @param $exportMode
+   * @param null $saveFile
+   * @param string $batchItems
+   */
+  public static function writeCSVFromTable($exportTempTable, $headerRows, $sqlColumns, $exportMode, $saveFile = NULL, $batchItems = '') {
+    $writeHeader = TRUE;
+    $offset = 0;
+    $limit = CRM_Export_BAO_Export::EXPORT_ROW_COUNT;
+    $ei = CRM_Hrjobcontract_ExportImportValuesConverter::singleton();
+
+    $query = "SELECT * FROM $exportTempTable";
+
+    while (1) {
```

Comment:
I know this is based on civicrm class , but civicrm implementation looks a bit weird to me and I think we are not compelled to follow what their class do  .. so  what if for some reason the break; statement never reached ? well , we will have a nice infinite loop ... I think it would be better to query the database to get the full rows count with :
$totalCount = executeQuery(SELECT COUNT(*) FROM $exportTempTable");

and then Then :

$loopCount = $totalCount/limit; 

and finally you can use :

while($loopCount) {
// logic here 
}